docs/website/plugins/nikita-packages/gatsby-node.js

```diff
@@ -0,0 +1,126 @@
+const constants = require('fs').constants
+
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions
+  createTypes(`
+    """
+    NikitaPackage Node
+    """
+    type NikitaPackage implements Node @infer {
+      name: String!
+      shortName: String!
+      slug: String!
+      version: String
+      edit_url: String!
+      frontmatter: Frontmatter
+    }
+    type Frontmatter {
+      navtitle: String
+      description: String
+      keywords: [String]
+    }
+  `)
+}
+
+exports.sourceNodes = async ({ actions, createNodeId }, pluginOptions) => {
+  const { createNode } = actions
+  // Get all packages
+  const files = await fs.readdir(pluginOptions.packagesDirectory)
+  files.forEach(async (file) => {
+    // Filter non-packages
+    const stat = await fs.stat(path.join(pluginOptions.packagesDirectory, file))
+    if(stat.isFile() || file == 'nikita') return
+    // Get package.json
+    try {
+      const packageJsonPath = path.join(pluginOptions.packagesDirectory, file, 'package.json')
+      await fs.access(packageJsonPath, constants.R_OK)  // check if exists
+      // Custom fields
+      packageJson = require(packageJsonPath)
+      const shortName = packageJson.name.replace('@nikitajs/', '')
+      const slug = `/current/actions/${shortName}/`
+      const edit_url = `https://github.com/adaltas/node-nikita/edit/master/packages/${shortName}/README.md`
+      // Create node
+      createNode({
+        // Custom fields
+        name: packageJson.name,
+        shortName: shortName,
+        slug: slug,
+        version: packageJson.version,
+        edit_url: edit_url,
+        frontmatter: {
+          navtitle: shortName,
+          description: packageJson.description,
+          keywords: packageJson.keywords,
+        },
+        // Gatsby fields
+        id: createNodeId(packageJson.name),
+        children: [],
+        internal: {
+          type: `NikitaPackage`,
+          // // An optional field. This is rarely used. It is used when a source plugin sources data it doesn’t know how to transform 
+          // content: content,
+          // the digest for the content of this node. Helps Gatsby avoid doing extra work on data that hasn’t changed.
+          contentDigest: crypto
+            .createHash(`md5`)
+            .update(packageJson.name)
+            .digest(`hex`)
+        }
+      })
+    } catch (err) {
+      console.error(err)
+    }
+  })
+}
+
+exports.createResolvers = ({ createResolvers, createNodeId }) => {
+  createResolvers({
+    NikitaPackage: {
+      readme: {
+        type: "Mdx",
+        async resolve(source, args, context, info) {
+          const res = await context.nodeModel.runQuery({
+            query: {
+              filter: {
+                slug: {eq: `${source.shortName}/README`}
+              },
+            },
+            type: "Mdx",
+            firstOnly: false,
+          })
+          return res[0]
+        }
+      }
+    }
+  })
+}
```

Comment:
The readme information are part of your NikitaPackage type, they can potentially contains metadata

Reply:
so far, there are `title`, `titleHtml`, `tableOfContents` and `body`. To put `frontmatter` of Readme to `NikitaPackage.frontmatter` and keep `body`, an additional request is needed:
```
  createResolvers({
    NikitaPackage: {
      readme: {
        type: "Mdx",
        async resolve(source, args, context, info) {
          const res = await context.nodeModel.runQuery({
            query: {
              filter: {
                slug: {eq: `${source.shortName}/README`}
              },
            },
            type: "Mdx",
            firstOnly: false,
          })
          return res[0]
        }
      },
      frontmatter: {
        async resolve(source, args, context, info) {
          const res = await context.nodeModel.runQuery({
            query: {
              filter: {
                slug: {eq: `${source.shortName}/README`}
              },
            },
            type: "Mdx",
            firstOnly: false,
          })
          return {
            ...res[0] !== undefined ? res[0].frontmatter : null,
            ...source.frontmatter
          }
        }
      }
    }
  })
```
I didn't find any other way.